bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/presence/NAPresenceCameraHandler.java

```diff
@@ -0,0 +1,176 @@
+            case CHANNEL_CAMERA_FLOODLIGHT:
+                return getFloodlightState();
+            case CHANNEL_CAMERA_FLOODLIGHT_AUTO_MODE:
+                //The auto-mode state shouldn't be updated, because this isn't a dedicated information. When the
+                // floodlight is switched on the state within the Netatmo API is "on" and the information if the previous
+                // state was "auto" instead of "off" is lost... Therefore the binding handles its own auto-mode state.
+                if (UnDefType.UNDEF.equals(floodlightAutoModeState)) {
+                    floodlightAutoModeState = getFloodlightAutoModeState();
+                }
+                return floodlightAutoModeState;
+        }
+        return super.getNAThingProperty(channelId);
+    }
+
+    private State getFloodlightState() {
+        if (module != null) {
+            final boolean isOn = NAWelcomeCamera.LightModeStatusEnum.ON.equals(module.getLightModeStatus());
+            return toOnOffType(isOn);
+        }
+        return UnDefType.UNDEF;
+    }
+
+    private State getFloodlightAutoModeState() {
+        if (module != null) {
+            return toOnOffType(NAWelcomeCamera.LightModeStatusEnum.AUTO.equals(module.getLightModeStatus()));
+        }
+        return UnDefType.UNDEF;
+    }
+
+    private void switchFloodlight(boolean isOn) {
+        if (isOn) {
+            changeFloodlightMode(NAWelcomeCamera.LightModeStatusEnum.ON);
+        } else {
+            switchFloodlightAutoMode(OnOffType.ON.equals(floodlightAutoModeState));
+        }
+    }
+
+    private void switchFloodlightAutoMode(boolean isAutoMode) {
+        floodlightAutoModeState = toOnOffType(isAutoMode);
+        if (isAutoMode) {
+            changeFloodlightMode(NAWelcomeCamera.LightModeStatusEnum.AUTO);
+        } else {
+            changeFloodlightMode(NAWelcomeCamera.LightModeStatusEnum.OFF);
+        }
+    }
+
+    private void changeFloodlightMode(NAWelcomeCamera.LightModeStatusEnum mode) {
+        Optional<String> localCameraURL = getLocalCameraURL();
+        if (localCameraURL.isPresent()) {
+            String url = localCameraURL.get()
+                    + FLOODLIGHT_SET_URL_PATH
+                    + "?config=%7B%22mode%22:%22"
+                    + mode.toString()
+                    + "%22%7D";
+            executeGETRequest(url);
+        }
+    }
+
+    private Optional<String> getLocalCameraURL() {
+        String vpnURL = getVpnUrl();
+        if (vpnURL != null) {
+            //The local address is (re-)requested when it wasn't already determined or when the vpn address was changed.
+            if (!cameraAddress.isPresent() || cameraAddress.get().isVpnURLChanged(vpnURL)) {
+                Optional<JSONObject> json = executeGETRequestJSON(vpnURL + PING_URL_PATH);
+                cameraAddress = json.map(j -> j.getString("local_url"))
+                        .map(localURL -> new CameraAddress(vpnURL, localURL));
```

Comment:
Using `getString` with throw an exception if it fails to find anything at that field name, use `optString` instead so it will return null.

Reply:
Good argument and solution. Thank you.

It is now fixed.